src/clj_http/core.clj

```diff
@@ -202,28 +202,30 @@
     (when max-redirects (.setMaxRedirects config max-redirects))
     (.build config)))
 
-(defmulti ^:private construct-http-host (fn [proxy-host proxy-port]
+(defmulti ^:private construct-http-host (fn [proxy-host proxy-port proxy-scheme]
                                           (class proxy-host)))
 (defmethod construct-http-host String
-  [^String proxy-host ^Long proxy-port]
-  (if proxy-port
-    (HttpHost. proxy-host proxy-port)
-    (HttpHost. proxy-host)))
+  [^String proxy-host ^Long proxy-port ^String proxy-scheme]
+  (cond
+    (and proxy-host proxy-port proxy-scheme) (HttpHost. proxy-host proxy-port proxy-scheme)
+    (and proxy-host proxy-port) (HttpHost. proxy-host proxy-port)
+    :else (HttpHost. proxy-host)))
 (defmethod construct-http-host java.net.InetAddress
-  [^InetAddress proxy-host ^Long proxy-port]
-  (if proxy-port
-    (HttpHost. proxy-host proxy-port)
-    (HttpHost. proxy-host)))
+  [^InetAddress proxy-host ^Long proxy-port ^String proxy-scheme]
+  (cond
+    (and proxy-host proxy-port proxy-scheme) (HttpHost. proxy-host proxy-port proxy-scheme)
+    (and proxy-host proxy-port) (HttpHost. proxy-host proxy-port)
+    :else (HttpHost. proxy-host)))
 
 (defn ^HttpRoutePlanner get-route-planner
```

Comment:
Is `get-route-planner` part of the public API for clj-http? This would be a breaking change if so.

Reply:
I don't see it in the documentation, and searching github for `get-route-planner` doesn't bring up anything except this repo, so I'm assuming it's not part of the public API.